connect/mirror/src/test/java/org/apache/kafka/connect/mirror/integration/MirrorConnectorsIntegrationBaseTest.java

```diff
@@ -770,6 +771,9 @@ public void testSyncTopicConfigs() throws InterruptedException {
         Map<ConfigResource, Collection<AlterConfigOp>> configOps = Collections.singletonMap(configResource, ops);
         // alter configs on target cluster
         backup.kafka().incrementalAlterConfigs(configOps);
+        // wait until the configs are changed
+        waitForConfigValueChange(backup, backupTopic, "delete.retention.ms", "2000");
+        waitForConfigValueChange(backup, backupTopic, "retention.bytes", "2000");
```

Comment:
thanks looking into this @atu-sharm !

I wonder if a wait for `retention.bytes` to be equal to 2000 will always succeed.
waitForCondition has a default poll interval of 100 ms. If the value is set to 2000 and then changed to 1000 by MirrorMaker within the time the thread sleeps (100ms) , we may miss the value change and timeout.

Can't we just wait on the futures returned by incrementalAlterConfigs ?

Reply:
@ashwinpankaj I looked into the suggestion to wait on the returned futures as it seemed quite promising, but it turns out that we already do that in the body of `EmbeddedKafkaCluster::incrementalAlterConfigs`: https://github.com/apache/kafka/blob/587f50d48f832f2ce76ed8f47b6ea472da3b0848/connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedKafkaCluster.java#L368-L370

Because of this, though, I'm pretty skeptical that any extra assertions about config changes making it to the backup cluster are going to be beneficial. We also perform assertions about propagated config updates inside a `waitForCondition` call, so there should already be bounded retry logic to handle potential delays in propagation. Maybe we just don't give enough time for that to work on Jenkins?

@atu-sharm have you considered adding more informative error messages to the [currently-failing assertion](https://github.com/apache/kafka/blob/587f50d48f832f2ce76ed8f47b6ea472da3b0848/connect/mirror/src/test/java/org/apache/kafka/connect/mirror/integration/MirrorConnectorsIntegrationBaseTest.java#L775-L789)? That may be useful for distinguishing between failures caused by legitimate bugs, bad testing logic, and timeouts that are too short for our CI machinery.

Reply:
Thanks, @ashwinpankaj, @C0urante, and @mimaison for taking the time to look into this. I agree with @C0urante. Since we have waited for all the results to come in and backup.kafka().incrementalAlterConfigs(configOps) didn't fail with any exceptions, further assertions to confirm that the configs have been set may not be beneficial.

Currently, the test fails with the delete.retention.ms value as 86400000 (which is the default value). As I'm new here, could you please provide more insights? There are multiple tests (namely testSyncTopicConfigs, testReplicateSourceDefault, testReplicateTargetDefault) and multiple classes extending this test class (IdentityReplicationIntegrationTest, MirrorConnectorsIntegrationExactlyOnceTest, etc). Is it possible that these tests, which are calling MM or altering the same topic config and if are running on different Gradle executors, could encounter a race condition and cause the tests to fail?

Reply:
As of now, i have added a lock for running the testcases which involves altering of same topic configs, your thoughts @C0urante @ashwinpankaj @mimaison